core/services/workflows/syncer/engine_registry.go

```diff
@@ -3,30 +3,36 @@ package syncer
 import (
 	"errors"
 	"sync"
-
-	"github.com/smartcontractkit/chainlink/v2/core/services/workflows"
 )
 
+// IsReadyCloser is an abstraction for engines that can be checked for readiness and closed.
+type IsReadyCloser interface {
```

Comment:
Nit: ReadyCloser is slightly more idiomatic in naming; in my other PR I pass in a service instead though